src/common/include/macros.fpp

```diff
@@ -117,3 +117,19 @@
 
 #define t_vec3   real(kind(0d0)), dimension(1:3)
 #define t_mat4x4 real(kind(0d0)), dimension(1:4,1:4)
+
+#:def ASSERT(predicate, message = None)
```

Comment:
this seems perhaps useful but you only use it in a few places in the code (for chemistry). can you perhaps at least point out where else it should be used in the code so that is makes sense to keep?

Reply:
Sure, it can clean up so much code. Take this example adapted from `m_checker` (there are literally hundreds of lines like this):
```fortran
if (acoustic(j)%npulse >= 5 .and. acoustic(j)%dipole) then
    call s_mpi_abort('acoustic('//trim(jStr)//')%dipole is not '// &
                     'supported for support >= 5 '// &
                     '(non-planar supports). Exiting ...')
end if
```

This can be simplified to:
```
@:PROHIBIT(acoustic(j)%npulse >= 5 .and. acoustic(j)%dipole, 'for dipole '//trim(jStr))
```

This is what would get printed (I actually ran this):
```
m_checker.fpp:186: Prohibited condition: acoustic(j)%npulse >= 5 .and. acoustic(j)%dipole. for dipole 1
```

I think it's much better than our current solution. Less code and less duplication. Of course you can still make this example better but I hope this proves my point.

Reply:
#591 🚀